### PR TITLE
[CLOUDGA-13540][CLOUDGA-13541]: Added GCP CMK support to create cluster and update CMK API

### PR DESCRIPTION
**Pull request by arishta-yb**
Summary:

Added GCP CMK support to these APIs:
1. Create cluster 
2. Update cluster CMK

Testing:

Tested cluster creation with CMK config for AWS and GCP providers via YBM CLI.
Tested update CMK API for both providers.

### CODE REVIEW

**Comment by shinji62** [comment]
Please fix the conflict first.

**Comment by arishta-yb** [comment]
> [shinji62](/shinji62)

Resolved the conflicts.

**Comment by cdavid** [comment]
Can you please do a `make doc` before merging?

File: cmd/cluster/encryption/encryptionutil.go [hunk 1]
```diff
@@ -25,14 +26,29 @@ import (
 	"golang.org/x/term"
 )
 
+func parseKMSKeyResourceID(resourceID string) (keyRingName string, keyName string, location string, protectionLevel string) {
+	parts := strings.Split(resourceID, "/")
+	if len(parts) != 8 || parts[0] != "projects" || parts[2] != "locations" || parts[4] != "keyRings" || parts[6] != "cryptoKeys" {
+		logrus.Fatalln("Invalid resource ID")
```
Comment (cdavid) [inline cmd/cluster/encryption/encryptionutil.go:32]: Can you please print the expected format of the key here?

File: cmd/cluster/encryption/encryptionutil.go [hunk 2]
```diff
@@ -58,47 +74,78 @@ func GetCmkSpecFromCommand(cmd *cobra.Command) (*ybmclient.CMKSpec, error) {
 				if len(strings.TrimSpace(val)) != 0 {
 					cmkAwsArnList = append(cmkAwsArnList, val)
 				}
+			case "resource-id":
```
Comment (cdavid) [inline cmd/cluster/encryption/encryptionutil.go:77]: gcp-resource-id

File: cmd/cluster/encryption/encryptionutil.go [hunk 3]
```diff
@@ -58,47 +74,78 @@ func GetCmkSpecFromCommand(cmd *cobra.Command) (*ybmclient.CMKSpec, error) {
...
+					// Validate non-empty key
+					if strings.TrimSpace(cmkAwsSecretKey) == "" {
+						logrus.Fatalln("The AWS Secret Key cannot be empty")
+					}
 				}
 			}
+			cmkSpec.AwsCmkSpec.Set(ybmclient.NewAWSCMKSpec(cmkAwsAccessKey, cmkAwsSecretKey, cmkAwsArnList))
+		case "GCP":
+			if cmkGcpResourceId == "" {
+				logrus.Fatalln("Incorrect format in CMK spec: GCP provider specified, but no resource-id provided")
```
Comment (cdavid) [inline cmd/cluster/encryption/encryptionutil.go:126]: gcp-resource-id

File: cmd/cluster/encryption/encryptionutil.go [hunk 4]
```diff
@@ -58,47 +74,78 @@ func GetCmkSpecFromCommand(cmd *cobra.Command) (*ybmclient.CMKSpec, error) {
...
+			}
+			keyRingName, keyName, location, protectionLevel := parseKMSKeyResourceID(cmkGcpResourceId)
+			if cmkGcpServiceAccountPath == "" {
+				logrus.Fatalln("Incorrect format in CMK spec: GCP provider specified, but no service-account-path provided")
```
Comment (cdavid) [inline cmd/cluster/encryption/encryptionutil.go:130]: gcp-service-account-path

**Comment by arishta-yb** [comment]
> Can you please do a `make doc` before merging?

Yeah sure

File: cmd/cluster/encryption/encryption.go [hunk 5]
```diff
@@ -82,6 +82,8 @@ var updateCmk = &cobra.Command{
 		authApi.GetInfo("", "")
 
 		clusterName, _ := cmd.Flags().GetString("cluster-name")
+		cloudProvider, _ := cmd.Flags().GetString("cloud-provider")
```
Comment (cdavid) [inline cmd/cluster/encryption/encryption.go:85]: This seems incorrect.

1. this is an invalid flag for the command (the command only takes an encryption spec)
2. the cloud-provider flag is usually the cloud provider of the cluster, not of the CMK (you can have GCP clusters with AWS CMK).

Comment (cdavid) [reply cmd/cluster/encryption/encryption.go:85]: You probably want to use `newCmkSpec.CloudProvider` or something equivalent in the if condition.

Comment (arishta-yb) [reply cmd/cluster/encryption/encryption.go]: Yes, you are correct, updated to fetch the cmk provider from newCmkSpec.

**Review by cdavid — APPROVED** [review]

**Pull request merged**